### DISCUSSION TITLE
Hi ! I have a question about the keystroke sequence in macros

### DISCUSSION THREAD

**Discussion by califodream** [macvim-dev/macvim#1401] [Q&A]
I use macros to record my keystroke sequence for completing some simple task. But the keystroke sequence is unreadable. Is there any skillful guy know what's going on out there. How to fix it. I got some screenshot here
<img width="404" alt="Screen Shot 2023-04-02 at 15 06 34" src="https://user-images.githubusercontent.com/62502402/229338060-22a66b3e-ecc1-41a8-a7e1-9415020feaa0.png">
You can see the sequence in register a where I record macros in. I don't know what the character mean of ^J, ^R etc
I want it can successfully map the keystroke like \<CR>, \<Esc>. etc. but these unreadable character. Please🤔

**Comment by ychin** [comment]
Why would you need to be able to read them? The standard usage is you use the `q` command to record macro, then you can play back using `@`. Vim records these keys and those character sequences are what it recorded (e.g. ^J is Ctrl-J). Are you trying to manually edit the macro content?  I'm not understanding what problem you are trying to solve exactly here.

**Comment by eirnym** [comment]
As MacVim is based on Vim, I'd suggest ask this question in Vim project. There you definitely get an answer on this particular question.
